### PR TITLE
add nur-update service

### PR DESCRIPTION
**Pull request by Mic92**
Add github token for [nur-update-server](https://github.com/nix-community/nur-update/):

```
$ nix build github:nix-community/nur-update
$ GITHUB_TOKEN=ghp_XXXXXX ./result/bin/nur-update-server
```

### CODE REVIEW

**Comment by zowoq** [comment]
I'll leave this open and push to it, should get it done in soon (day or two).

**Comment by Mic92** [comment]
DynamicUser should be fine.

File: services/nur-update/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    forceSSL = true;
+    locations."/".proxyPass = "http://unix:/run/nur-update/gunicorn.sock";
+  };
+
+  sops.secrets.nur-update-github-token = { };
+
+  systemd.services.nur-update = {
+    description = "nur-update service";
+    serviceConfig = {
+      ExecStart = ''
```
Comment (Mic92) [inline services/nur-update/default.nix:15]:
```suggestion
      DynamicUser = true;
      ExecStart = ''
```

Comment (zowoq) [reply services/nur-update/default.nix:16]: Done.

File: services/nur-update/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        GITHUB_TOKEN="$(<$CREDENTIALS_DIRECTORY/github-token)" \
+          ${lib.getExe pkgs.python3.pkgs.gunicorn} nur_update:app \
+          --bind unix:/run/nur-update/gunicorn.sock \
+          --log-level info \
+          --python-path ${nur-update.packages.${pkgs.system}.default} \
+          --timeout 30 \
+          --workers 3
+      '';
+      LoadCredential = [ "github-token:${config.sops.secrets.nur-update-github-token.path}" ];
+      Restart = "always";
```
Comment (Mic92) [inline services/nur-update/default.nix:26]: This should be required for the socket to work:

```suggestion
      Restart = "always";
      RuntimeDirectory = "nur-update";
```

Comment (zowoq) [reply services/nur-update/default.nix:26]: Always forget that.

**Comment by zowoq** [comment]
Moved dns to another PR that can be merged first so I can test this works on build03 before merging it.

**Pull request merged**